core/backend/src/ChangesetECAdaptor.ts

```diff
@@ -404,22 +405,234 @@ namespace DateTime {
+   * Sets the value associated with the specified key in the cache.
+   * @param key - The key to set the value for.
+   * @param value - The value to be associated with the key.
+   */
+  set(key: string, value: ChangedECInstance): void;
+
+  /**
+   * Returns an iterator for all the values in the cache.
+   * @returns An iterator for all the values in the cache.
+   */
+  all(): IterableIterator<ChangedECInstance>;
+
+  /**
+   * Returns the number of entries in the cache.
+   * @returns The number of entries in the cache.
+   */
+  count(): number;
+}
+
+/**
+ * In-memory cache for storing changed EC instances.
+ * @beta
+ */
+export class InMemoryInstanceCache implements ECChangeUnifierCache {
+  private readonly _cache = new Map<string, ChangedECInstance>();
+
+  /**
+   * Retrieves the changed EC instance associated with the specified key.
+   * @param key - The key used to retrieve the instance.
+   * @returns The changed EC instance, or undefined if not found.
+   */
+  public get(key: string): ChangedECInstance | undefined {
+    return this._cache.get(key);
+  }
+
+  /**
+   * Sets the changed EC instance associated with the specified key.
+   * @param key - The key used to store the instance.
+   * @param value - The changed EC instance to be stored.
+   */
+  public set(key: string, value: ChangedECInstance): void {
+    const meta = value.$meta as any;
+    if (meta) {
+      Object.keys(meta).forEach(key => meta[key] === undefined && delete meta[key]);
+    }
+    this._cache.set(key, value);
+  }
+
+  /**
+   * Returns an iterator over all the changed EC instances in the cache.
+   * @returns An iterator over all the changed EC instances.
+   */
+  public *all(): IterableIterator<ChangedECInstance> {
+    for (const key of Array.from(this._cache.keys()).sort())
+      yield this._cache.get(key)!;
+  }
+
+  /**
+   * Returns the number of changed EC instances in the cache.
+   * @returns The number of changed EC instances.
+   */
+  public count(): number {
+    return this._cache.size;
+  }
+
+  /**
+   * Disposes the cache.
+   */
+  public [Symbol.dispose](): void {
+    // Implementation details
+  }
+}
+
+/**
+ * Represents a cache for unifying EC changes in a SQLite-backed instance cache.
+ * @beta
+ */
+export class SqliteBackedInstanceCache implements ECChangeUnifierCache {
+  /**
+   * Creates an instance of SqliteBackedInstanceCache.
+   * @param _db The underlying database connection.
+   * @param bufferedReadInstanceSizeInBytes The size of each read instance in bytes for buffering.
+   * @throws Error if bufferedReadInstanceSizeInBytes is less than or equal to 0.
+   */
+  public constructor(private readonly _db: AnyDb, public readonly bufferedReadInstanceSizeInBytes: number = 1024 * 1024 * 10) {
+    if (bufferedReadInstanceSizeInBytes <= 0)
+      throw new Error("bufferedReadInstanceCount must be greater than 0");
+    this.createTempTable();
+  }
+
+  private readonly _cacheTable = `[temp].[${Guid.createValue()}]`;
```

Comment:
```suggestion
  // Table name should be unique in case two caches are instantiated
  private readonly _cacheTable = `[temp].[${Guid.createValue()}]`;
```